enterprise_gateway/services/processproxies/k8s.py

```diff
@@ -310,14 +326,66 @@ def _create_kernel_namespace(self, service_account_name: str) -> str:
                     body = client.V1DeleteOptions(
                         grace_period_seconds=0, propagation_policy="Background"
                     )
-                    client.CoreV1Api().delete_namespace(name=namespace, body=body)
+                    client.CoreV1Api(api_client=kubernetes_client).delete_namespace(
+                        name=namespace, body=body
+                    )
                     self.log.warning(f"Deleted kernel namespace: {namespace}")
                 else:
                     reason = f"Error occurred creating namespace '{namespace}': {err}"
                 self.log_and_raise(http_status_code=500, reason=reason)
 
         return namespace
 
+    def _create_service_account_if_not_exists(
```

Comment:
If this is strictly for external clusters, could we rename this to something like: `_create_external_service_account_if_not_exists` or `_create_remote_service_account_if_not_exists`.  Since the other configurable options refer to "external", I guess I'd prefer the former.